Sources/JWT/Signers/RSASigner+X509.swift

```diff
@@ -0,0 +1,30 @@
+import CTLS
+
+typealias CX509Key = UnsafeMutablePointer<X509>
+
+extension RSAKey {
+    public init(x509Cert cert: Bytes) throws {
+        guard
+            let cert = cert.withUnsafeBufferPointer({ rawKeyPointer -> CX509Key? in
```

Comment:
Sounds great in theory but I'm struggling to find a good place to put the defer block.

In the guard statement, let's say assigning to `cert` succeeds but the `pubkey` line fails. How can you clean up after that? There's no access to `cert` in the `else` block. Does this mean we'll need 3 separate guard blocks, with 2 defer blocks cleaning up the `cert` and `pubKey`?